src/query/frontend/semantic/symbol_generator.cpp

```diff
@@ -157,25 +157,33 @@ void SymbolGenerator::VisitReturnBody(ReturnBody &body, Where *where) {
 // Query
 
 bool SymbolGenerator::PreVisit(SingleQuery &) {
-  prev_return_names_ = curr_return_names_;
-  curr_return_names_.clear();
+  auto &scope = scopes_.back();
+
+  scope.prev_return_names = scope.curr_return_names;
+  scope.curr_return_names.clear();
   return true;
 }
 
 // Union
 
 bool SymbolGenerator::PreVisit(CypherUnion &) {
+  auto scope = scopes_.back();
   scopes_.back() = Scope();
```

Comment:
I am not sure this is correct? But I would suggest to use emplace not directly referencing back()
```suggestion
  scopes_.emplace_back(Scope());
```

Reply:
Actually this scopes_.back() = Scope() is legacy. They wanted to just clear the scopes_ and add a new scope